### DISCUSSION TITLE
Replace default `height` for "banner image" in grid listing by `max-height`?

### DISCUSSION THREAD

**Discussion by mcanouil** [quarto-dev/quarto-cli#2990] [Feature Requests]
Hi,
currently, the grid listing has a default height of 150px for top image in the card which causes the image to be "cropped" in nearly all situations.
Instead, I would suggest to not set `height` property but`max-height` to allow responsive scaling of the image without cropping the image.
For example, on the left, manually defined max-height, while on the right it's the default from the grid listing.
<img width="741" alt="Screenshot of a grid listing made with quarto using a three columns layout and showing two cards. On the left a card with an image properly scaled without cropping and on the right a image vertically cropped due to css height property set to 150 pixels" src="https://user-images.githubusercontent.com/8896044/197394786-d40b57a1-bad5-456c-bd76-8c523765f329.png">

A video to show the resizing issue of setting `height`.

https://user-images.githubusercontent.com/8896044/197395474-62bf6851-1f4e-47bc-8f8c-0b067c1e79f4.mov

**Comment by dragonstyle** [comment]
I definitely think it is worth considering making our grid more flexible. The `height` is currently set intentionally to make the grid consistent in size and the title and other document data consistently aligned. But I agree that there are cases where this isn't desirable. I'll target for 1.3 and we can take a closer look...

**Reply by mcanouil** [reply]
I also like the aligned part of the grid, this is why all my cover images are 16/9 ratio, but setting height while width is resized is not great.
Maybe an option for grid listing allowing or not the height to be fixed, e.g., `fixed-height: true` (default) 🤔
I'll be happy to share some thoughts on this later when you'll start working on this.

**Reply by dragonstyle** [reply]
Yeah that sounds very good. We could also consider exposing the `object-fit` which is controlling how under or oversized images are scaled...

**Reply by mcanouil** [reply]
There is a JavaScript library that could be nice as well and related to let the cards in grid layout have flexible height.
The [Mansory.js](https://masonry.desandro.com/) library.
I am working on a Quarto extension (<https://github.com/mcanouil/quarto-masonry>), but it requires a bit of work to actually make it work properly with grid listings.

**Reply by dragonstyle** [reply]
That is really interesting. It is almost like it should be another grid layout altogether. I have it on my todo list to make grid layouts extensible using extensions- it would be awesome to request a listing layout `masonry` and get this type of thing...

**Reply by mcanouil** [reply]
It might be actually a better idea to make a new layout option with no `height` set and the use of Masonry.js, and let "grid" be a fixed grid layout. 👍

**Reply by dragonstyle** [reply]
Yeah I sort of like that approach. Masonry looks very cool for making really compact layouts of wildly different grid cards which is both very cool but perhaps different enough than our simple grid to warrant a new layout type!

**Comment by mcanouil** [comment]
Closing this in favour of #2999
